flux_led/__main__.py

```diff
@@ -574,32 +576,68 @@ def _determineMode(self, ww_level, pattern_code):
                 mode = "color"
         elif pattern_code == 0x60:
             mode = "custom"
+        elif pattern_code == 0x41:
+            mode = "color"
         elif PresetPattern.valid(pattern_code):
             mode = "preset"
         elif BuiltInTimer.valid(pattern_code):
             mode = BuiltInTimer.valtostr(pattern_code)
 
         return mode
 
-    def update_state(self, retry=2):
+    def query_state(self, retry=2, led_type = None):
+
+        # default values
         msg = bytearray([0x81, 0x8a, 0x8b])
+        msg_len = 14
+
+        if self.protocol == 'LEDnet_wifi370' or led_type == 'LEDnet_wifi370':
+            msg =  bytearray([0xef, 0x01, 0x77])
+            msg_len = 11
+            led_type = 'LEDnet_wifi370'
+            self._use_csum = False
         try:
             self._send_msg(msg)
-            rx = self._read_msg(14)
+            rx = self._read_msg(msg_len)
```

Comment:
If you don't specify that the original protocol should be used, we'll try to read 14 bytes even though the device will only send 11. The device is sending us enough information to let us know how much data it'll send, so we should figure that out automatically rather than requiring manual configuration.